### PR TITLE
Fix/crash in map snapshotter activity

### PR DESCRIPTION
**Pull request by ovivoda**
Fix MapSnapshotterActivity.kt crash if there is IllegalStateException exception thrown from native code.

However I think we will need to dig in a bit more to find out the real problem. 

cc @artakka @birkskyum Any clues on the root cause?

### CODE REVIEW

**Comment by alanchenboy** [comment]
Just curious, can be captured in native code? there  may throw exceptions in `mbgl::map::renderStill`

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.kt [hunk 1]
```diff
@@ -652,22 +666,26 @@ open class MapSnapshotter(context: Context, options: Options) {
...
+                            addLayerBelow(layerWrapper.getLayer(), layerWrapper.belowLayer)
+                        } else {
+                            addLayerBelow(layerWrapper.layer, MapboxConstants.LAYER_ID_ANNOTATIONS)
+                        }
+                    }
+                    for (image in builder.images) {
+                        addImage(image.id, image.bitmap, image.isSdf)
+                    }
+                } catch (e: IllegalStateException) {
+                    e.printStackTrace()
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.kt:688]: Did you left this in on purpose?

Comment (ovivoda) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.kt:688]: Yes. Just fixing the crash not the cause. Once the cause is fixed we can reuse fully previous code.

**Comment by ovivoda** [comment]
This is a temporary fix for showcasing the MapLibre features. We should investigate more the native cause of the crash.

**Comment by ovivoda** [comment]
> mbgl::map::renderStill

Indeed this should be the next step. To investigate more on this part of the native code. Not trivial to debug though but I will give it a try.

**Comment by ovivoda** [comment]
linking initial reported issue to this for reference: https://github.com/maplibre/maplibre-gl-native/issues/903

**Review by louwers — CHANGES_REQUESTED** [review]
Any quick fixes need to be documented with comments.

Can we reproduce the issue with a test?

**Comment by louwers** [comment]
Added a test but it's passing, so need to tweak it a bit more.

**Comment by louwers** [comment]
OK now the test reproduces the error (if the exception is not caught).

**Comment by louwers** [comment]
I am digging into JNI to be able debug this. But it might take a while.

We should add a comment to the catch block if this quick fix is urgent.

**Comment by ovivoda** [comment]
Actually I think I would go for another approach. I would leave `MapSnapshotter` in place as it was and create a failing test with the loaded style from JSON and fix the problem for that one.

**Comment by louwers** [comment]
> Actually I think I would go for another approach. I would leave `MapSnapshotterTest` in place as it was and create a failing named something like`MapSnapshotterLocalStyleLoadTest` test with the loaded style from JSON and fix the problem for that one. 

I can use another name but I would use the same file. The code for running the test is identical, it is just the `MapSnapshotter.Options` that is different.

**Comment by ovivoda** [comment]
I am closing this one as the main problem relies somewhere in the native code.

**Pull request closed without merging**